### PR TITLE
Webui destructive tests support

### PR DESCRIPTION
**Pull request by KKoukiou**
No need to review yet, I want to look at it again tomorrow with a clear head. Feel free of course if you want.

TODO:
* [x] Update test/common tag to fetch https://github.com/cockpit-project/cockpit/pull/16914
* [x] Fix the pylint issues

### CODE REVIEW

File: ui/webui/test/machine_install.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    def pull(self, _):
+        """Download iso.
+        """
+        image_file = os.path.join(WEBUI_TEST_DIR, "images", self.image)
+        if not os.path.exists(image_file):
+            try:
+                if self.image == 'fedora-rawhide':
+                    subprocess.Popen(
+                        f"curl -L https://download.fedoraproject.org/pub/fedora/linux/development/rawhide/Server/x86_64/os/images/boot.iso -o {image_file}",  # nopep8
```
Comment (KKoukiou) [inline ui/webui/test/machine_install.py:125]: Reminder to myself. Move this to a script as 'pull' runs only from tests-run

**Review by rvykydal — APPROVED** [review]
Looks good to me, thank you.
(The run-tests in the PR is just temporary, we will be using cocpit's run-tests updated to allow to use VirtInstallMachine introduced in this PR as GlobalMachine).

**Comment by KKoukiou** [comment]
Cleanly rebased.

File: ui/webui/test/machine_install.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        self._execute(
+            f"virsh -q -c qemu:///session undefine "
+            f"--remove-all-storage {self.label} || true"
+        )
+        if self.http_server:
+            self.http_server.kill()
+
+    def pull(self, _):
+        """Download iso.
+        """
```
Comment (jkonecny12) [inline ui/webui/test/machine_install.py:119]: Could you please make this a one line comment?

File: ui/webui/test/run-tests [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+        self.machine = machine_install.VirtInstallMachine(image=self.image)
+        if not os.path.exists(self.machine.image_file):
+            self.machine.pull(self.machine.image_file)
+        self.machine.start()
+
+    def kill(self):
+        self.machine.kill()
+
+
+def run(opts, image):
```
Comment (jkonecny12) [inline ui/webui/test/run-tests:199]: I would love to have this split into smaller functions. It's too big now.

**Review by jkonecny12 — COMMENTED** [review]
Looks great to me. I just have a few notes below.

File: ui/webui/test/run-tests (hunk 3 above)
Comment (KKoukiou) [reply ui/webui/test/run-tests:199]: This file as I mention in the start, is a copy paste from cockpit, with some sligh adjustments. The original script needs to be adjusted and this is about to be deleted soon.

Comment (jkonecny12) [reply ui/webui/test/run-tests:199]: Ahh, ok thanks for explanation.

**Review by VladimirSlavik — APPROVED** [review]
Thank you, I think it's good, as far as I can tell...

File: ui/webui/test/machine_install.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            for _ in range(10):
+                try:
+                    requests.get(
+                        f"http://{self.web_address}:{self.web_port}/"
+                        "cockpit/@localhost/anaconda-webui/index.html"
+                    )
+                    break
+                except requests.exceptions.RequestException:
+                    time.sleep(5)
+            else:
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply ui/webui/test/machine_install.py:102]: The for look can have an else in python :) Not often used.
https://book.pythontips.com/en/latest/for_-_else.html#else-clause

File: ui/webui/test/webui_testvm.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    machine.wait_boot()
+
+    # print ssh command
+    print("ssh -o ControlPath=%s -p %s %s@%s" %
+          (machine.ssh_master, machine.ssh_port, machine.ssh_user, machine.ssh_address))
+    # print Cockpit web address
+    print(
+        "http://%s:%s/cockpit/@localhost/anaconda-webui/index.html" %
+        (machine.web_address, machine.web_port)
+    )
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply ui/webui/test/webui_testvm.py:40]: Mind if we improve it in the followup with your README improvements? It's massive PR, would rather not delay it much more.

**Pull request merged**